### PR TITLE
Added 2.3.1 changes to changelog.

### PR DESCRIPTION
**Pull request by binary1248**

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -1,3 +1,51 @@
+SFML 2.3.1
+==========
+
+Also available on the website: http://www.sfml-dev.org/changelog.php#sfml-2.3.1
```
Comment (MarioLiebisch) [inline changelog.txt]: Would reword:

> This change log is also available...

Otherwise it seems like it refers to the actual library.

**Comment by Bromeon** [comment]
Shouldn't we focus on changes that have an impact on the user, either as fixed bugs or features?

There are minor trivialities like

> * Removed unnecessary sf:: prefixes (#888) - *completely internal*
* Corrected documentation typo in RenderStates.hpp (#889) - *good to know! :P*

which don't belong in a changelog in my opinion. I'm not even sure if I would mention any documentation improvements at all... If I as a user look at changelogs, I'm interested in functional changes, not internal cosmetics or docs.

File: changelog.txt (hunk 1 above)
Comment (Bromeon) [inline changelog.txt]: In the past we always used the wording "Also available...".

**Comment by binary1248** [comment]
So... why were non-functional changes included in the previous changelogs?

**Comment by Bromeon** [comment]
I don't know... They were much bigger, and those changes had an easy job to hide :D

Aside from older changelogs, what's your (and others') opinion on this?

**Comment by eXpl0it3r** [comment]
I don't really care. :smile: 
In the past, I tried to add one point in the general section that would just say that some documentation changes have happened and then link the various issues.

**Comment by MarioLiebisch** [comment]
Yeah, pick changes that make a difference. Documentation that's been wrong or erronous should be an exception IMO. Other than that I don't see the need to mention fixed typos outside example source, even though they've had their own pull request/commit.

**Comment by dabbertorres** [comment]
For whatever it may be worth, my recommendation would be to keep all
changes in the change logs, but sort them.
Put all the user facing changes and such first, under a label, like,
"Important", or " User-Facing", or something.
And then add all the little "unimportant" changes afterwards, labeled as
"Miscellaneous" or something.
That way an easy to read log is kept of every change.

**Comment by Bromeon** [comment]
> For whatever it may be worth, my recommendation would be to keep all
changes in the change logs, but sort them.

What kind of useful information do I have as a user if I know a typo has been fixed, or an internal syntax has been adapted? There's not a single good reason to include such changes in a public changelog...

You also keep other changes at a high-level description like "bug X fixed" and not "line X changed to Y". Users want to know how they're affected, and not what characters in the source are different. For that, we have the commit history.

**Comment by dabbertorres** [comment]
Ooh, I had a slight misunderstanding. My bad. In that case, I completely agree.

**Comment by eXpl0it3r** [comment]
This PR has been added to my merge list, meaning it will be merged into the 2.3.x branch soon, unless someone raises any concerns.

**Comment by Bromeon** [comment]
Hm... Are those not concerns? :wink:

**Comment by binary1248** [comment]
Given that most of these sorts of discussions end up in endless circles, it's kind of hard to tell if there are actually any _real_ concerns in the end. You know... the kind people might actually lose sleep over...

**Comment by eXpl0it3r** [comment]
There's always the possibility to create another PR and then we can decide. I'm just tired of waiting till somebody lifts their fingers and does something, only to watch said work being talked down without any counter action.

I mean sure, we can wait another few months till the changelog get written and we can release 2.3.1, who cares anyways?

**Comment by Bromeon** [comment]
> There's always the possibility to create another PR and then we can decide. 

**That** is how you create unnecessary bureaucratic overhead. Not by fixing things right now. Since we all agree (or don't care), I don't see the problem of an endless discussion. It seems like we already found consensus 2 days ago.

>  I'm just tired of waiting till somebody lifts their fingers and does something, only to watch said work being talked down without any counter action.

I wonder why we even have PRs if the preferred way is merging them unchanged and ignoring any ongoing discussion. That's easier achieved by pushing to master directly.

I did not provide a commit, because the guy who opens the PR usually applies the changes, and I don't like to push onto other people's branches/PRs without asking. But I'll gladly do it -- just tell me whether I should push a second commit, which is a bit exaggerated for such simple changes, or modify binary1248's commit with a sign-off.

**Comment by eXpl0it3r** [comment]
A PR is a Pull **Request**. There can be many PR for one particular feature, bugfix, enhancement, etc. We're not limited to one branch and one PR only.

SFML 2.3.1 has been waiting *weeks* for someone to write the changelogs.

**Comment by binary1248** [comment]
Removed all the non-functional stuff. Should be good to go now.

**Comment by Bromeon** [comment]
Thank you! :+1:

And about the discussion: I really didn't intend to do it overly long or just block the PR just for fun -- you know I'm not like that, I'm really interested in a good library. And I'd like to know how to proceed in the future... In my opinion, if a pull request is opened, a discussion is welcome, isn't it? For such simple changes, if we have something to correct, should we open yet another PR with our own version? I'm not a big fan of that, because 1. it will only spread the discussion to two places and make things more complicated, and 2. it looks like the effort of the original requester is "stolen"... 

Maybe we could create another branch though, push a commit on top and reference that branch in the original PR.
